pkg/serverconfig/types.go

```diff
@@ -65,6 +65,36 @@ type Customization struct {
 	DocumentationBaseURL string `yaml:"documentationBaseURL,omitempty"`
 	CustomProductName    string `yaml:"customProductName,omitempty"`
 	CustomLogoFile       string `yaml:"customLogoFile,omitempty"`
+	// console allow the cluster admin to configure the shown categories.
+	DeveloperCatalog DeveloperConsoleCatalogCustomization `yaml:"developerCatalog"`
+}
+
+// ConsoleCatalog allow the cluster admin to configure developer catalog.
+type DeveloperConsoleCatalogCustomization struct {
+	// categories which are shown the in the devconsole.
+	Categories []DeveloperConsoleCatalogCategory `json:"categories"`
+}
+
+type DeveloperConsoleCatalogCategory struct {
+	// identifier used in the URL to enable deeplinking in console
+	ID string `json:"id"`
+	// category display label
+	Label string `json:"label"`
+	// the list of item tags this category will match
+	// +optional
```

Comment:
I don't think these comments have any meaning in the console repository, only in openshift/api.